### PR TITLE
Simplify the number unboxing logic

### PR DESCRIPTION
**Pull request by lewing**
From the js binding perspective all of these types are numbers (see the method binding) so treat them that way.

### CODE REVIEW

**Comment by kg** [comment]
This stops being a good idea once i64/u64 support is added, though, because those aren't Number

**Comment by lewing** [comment]
longs will have to be handled specially in any case. This preserves the existing logic perfectly

**Review by kg — CHANGES_REQUESTED** [review]
Other than my comments, seems fine - let's wait until my bindings optimizations are landed before landing this since they will conflict.

File: src/mono/wasm/runtime/binding_support.js
Comment (lewing) [reply src/mono/wasm/runtime/binding_support.js:567]: No it doesn't do the same thing, it also isn't actually used.  I can fix it to do mostly the same thing as the current code but it isn't really testable at the moment.

File: src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs
Comment (lewing) [reply src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs:238]: I don't like it either, this works in more cases than the current code.

File: src/mono/wasm/runtime/binding_support.js
Comment (kg) [reply src/mono/wasm/runtime/binding_support.js:567]: If we're sure it's not used then I'm OK with this.

File: src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs
Comment (lewing) [reply src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs:238]: To clarify, this code is only called by the number boxing code on the js side which does an even less effective check

File: src/mono/wasm/runtime/binding_support.js
Comment (kjpou1) [reply src/mono/wasm/runtime/binding_support.js:567]: Yes not used right now.  PR being worked on to re enable this functionality

**Comment by lewing** [comment]
blocked against #41808 which is prefered

File: src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs
Comment (lewing) [reply src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs:234]: I'm fine with Truncate, but now I'm curious how the interpreter handles the two cases.

Comment (kg) [reply src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs:234]: Yeah, interesting point. I imagine Math.Truncate probably performs a fn call, but maybe it's an intrinsic? ```(int)d``` should also work with the caveat that it might do something slightly different for negative numbers, I'd have to check.

**Pull request closed without merging**
